packages/flutter/lib/src/gestures/tap_and_drag.dart

```diff
@@ -984,14 +997,24 @@ sealed class BaseTapAndDragGestureRecognizer extends OneSequenceGestureRecognize
 
     _wonArenaForPrimaryPointer = true;
 
-    // resolve(GestureDisposition.accepted) will be called when the [PointerMoveEvent] has
-    // moved a sufficient global distance.
-    if (_start != null) {
+    // resolve(GestureDisposition.accepted) will be called when the [PointerMoveEvent]
+    // has moved a sufficient global distance to be considered a drag and
+    // `eagerVictoryOnDrag` is set to `true`.
+    if (_start != null && eagerVictoryOnDrag) {
       assert(_dragState == _DragState.accepted);
       assert(currentUp == null);
       _acceptDrag(_start!);
     }
 
+    // This recognizer will wait until it is the last one in the gesture arena
+    // before accepting a drag when `eagerVictoryOnDrag` is set to `false`.
+    if (_start != null && !eagerVictoryOnDrag) {
+      assert(_dragState == _DragState.possible);
+      assert(currentUp == null);
+      _dragState = _DragState.accepted;
```

Comment:
Why is this not set in the previous `if` too?

Reply:
Because in the case of `eagerVictoryOnDrag == true`, it is set immediately when the drag is detected.